### ISSUE TITLE
Focus style for any keyboard operable control

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#574]
See WCAG 2.4.7 Focus Visible: Any keyboard operable user interface has a mode of operation where the keyboard focus indicator is visible. (Level AA)
https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-visible
https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-visible.html

See also the looong conversation on #127, one of the points discussed there is that there are several ways to meet this requirement but color alone may be not enough. In most of the cases an additional indicator, like a shape, is required unless the color change is very, very, evident like a complete inversion of colors (colors that already have a sufficient contrast ratio).

As inspiration, a couple examples:
- on #527 I'm proposing to just pair the inserter items hover and focus styles
- other controls could use the WordPress "circular focus", see below for a quick experiment

<img width="811" alt="circular focus experiment" src="https://cloud.githubusercontent.com/assets/1682452/25556971/da641c5a-2d07-11e7-8354-f2e96763bf61.png">

**Comment by afercia** [comment]
Note that the "Understanding SC 2.4.7" notes make clear this WCAG requirement is not intended just for keyboard users, it's for everyone.

**Comment by jasmussen** [comment]
Some work also being done in #904 and #942.

**Comment by mtias** [comment]
I am closing this one as most elements should have focus styles already, and any missing ones we should open specific issues for to keep it tidy.
